android/src/main/java/org/conscrypt/Platform.java

```diff
@@ -736,10 +736,14 @@ public static String getOriginalHostNameFromInetAddress(InetAddress addr) {
                 return originalHostName;
             } catch (InvocationTargetException e) {
                 throw new RuntimeException("Failed to get originalHostName", e);
-            } catch (ClassNotFoundException ignore) {
-                // passthrough and return addr.getHostAddress()
-            } catch (IllegalAccessException ignore) {
-            } catch (NoSuchMethodException ignore) {
+            } catch (RuntimeException e) {
+                throw e;
+            } catch (Exception ignore) {
+                // Everything else...
+                // Passthrough and return addr.getHostAddress()
+
+                // NOTE: We're really catching ReflectiveOperationException, which was added in
+                // Java 7. Catching a generic exception for backward-compatibility with Java 6.
```

Comment:
This wasn't catching ReflectiveOperationException (and it's in Android code).

Reply:
@flooey what's the right thing to do here? Why was Android not catching `ReflectiveOperationException`?